src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -193,6 +193,28 @@ bool convert_node_input_precision(const std::shared_ptr<ov::Node>& node,
     return false;
 }
 
+// Part of hotfix from CVS-155745 & CVS-155990
+static bool node_input_output_type_different(const std::shared_ptr<ov::Node> node, const precisions_map& precisions) {
+    for (const auto& p : precisions) {
+        const auto& type_from = p.first;
+
+        auto different_type_input = [&type_from](const ov::Input<Node>& input) -> bool {
+            return input.get_element_type() != type_from;
+        };
+
+        auto different_type_output = [&type_from](const ov::Output<Node>& output) -> bool {
+            return output.get_element_type() != type_from;
+        };
```

Comment:
these two functions are functionally equivalent. Please create just one `is_element_type_different` and use for both cases

Reply:
Done, using outputs only now